### PR TITLE
[link prober] print out error code message when `sendHeartbeat` fails

### PR DESCRIPTION
**Pull request by zjswhhh**
### Description of PR

Summary:
Fixes # (issue)
Print out error code message when `sendHeartbeat` fails.

sign-off: Jing Zhang zhangjing@microsoft.com 

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Doc/Design
- [ ] Unit test

##### Work item tracking
- Microsoft ADO **(number only)**:

### CODE REVIEW

File: src/link_prober/LinkProber.cpp [hunk 1]
```diff
@@ -333,7 +333,7 @@ void LinkProber::sendHeartbeat(bool forceSend)
         mStream.write_some(boost::asio::buffer(mTxBuffer.data(), mTxPacketSize), errorCode);
 
         if (errorCode) {
-            MUXLOGTRACE(mMuxPortConfig.getPortName() + ": Failed to send heartbeat!");
+            MUXLOGTRACE(mMuxPortConfig.getPortName() + ": Failed to send heartbeat! Error code: " + errorCode.message());
```
Comment (lolyu) [inline src/link_prober/LinkProber.cpp:336]: what's the output of `errorCode.message()`?

**Review by lolyu — COMMENTED** [review]
do we also need to catch the I/O error in the receive?

File: src/link_prober/LinkProber.cpp (hunk 1 above)
Comment (zjswhhh) [reply src/link_prober/LinkProber.cpp:336]: A string, i.e. for `boost::system::errc::bad_file_descriptor` it will return "Bad file descriptor"

**Review by lolyu — APPROVED** [review]
LGTM

**Pull request merged**